Angular/src/app/app.component.ts

```diff
@@ -0,0 +1,111 @@
+import {NgModule, Component, enableProdMode} from '@angular/core';
+import {BrowserModule} from '@angular/platform-browser';
+import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
+import {CustomSchedulerComponent} from "./components/custom-scheduler/custom-scheduler.component";
+import {FilterFormComponent} from "./components/filter-form/filter-form.component";
+import {DxSchedulerModule, DxFormModule} from 'devextreme-angular';
+import {OptionChangedEvent} from "devextreme/ui/form";
+import {startViewDate, endViewDate} from './config';
+import {Appointment, Service, data} from "./data";
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.css'],
+  providers: [Service],
+})
+
+export class AppComponent {
+  dataSource: Appointment[];
+  currentDate: Date = new Date(2022, 9, 1);
+  useDisable: boolean = false;
+  startDate: Date = startViewDate;
+  endDate: Date = endViewDate;
+  filterValue: string = '';
+
+  constructor(service: Service) {
+    this.dataSource = service.getAppointments();
+  }
+
+  onFilterValueChange({event}: any) {
+    this.filterValue = event.currentTarget.value.toLowerCase();
+
+    this.dataSource = this.createFilteredAppointments(
+      this.startDate,
+      this.endDate,
+      this.filterValue,
+      this.useDisable
+    )
+  }
+
+  onCheckboxChange() {
+    this.useDisable = !this.useDisable;
+
+    this.dataSource = this.createFilteredAppointments(
+      this.startDate,
+      this.endDate,
+      this.filterValue,
+      this.useDisable
+    )
+  }
+
+  onStartDateChange(e: OptionChangedEvent) {
+    this.startDate = e.value;
+
+    this.dataSource = this.createFilteredAppointments(
+      this.startDate,
+      this.endDate,
+      this.filterValue,
+      this.useDisable
+    )
+  }
+
+  onEndDateChange(e: OptionChangedEvent) {
+    this.endDate = e.value;
+
+    this.dataSource = this.createFilteredAppointments(
+      this.startDate,
+      this.endDate,
+      this.filterValue,
+      this.useDisable
+    )
+  }
+
+  filterAppointments(startDate: Date, endDate: Date, filterValue: string, appointment: Appointment) {
```

Comment:
Let's make these methods private, rename and reorder them.
Also, we can reduce the count of the arguments of these methods with FiltersValue interface :)

```ts
  private createFilteredAppointments(filterValues: FilterValues) {
    const { useDisable} = filterValues;
    if (useDisable)
      return data.map(appointment => ({
        ...appointment,
        disabled: !this.isAppointmentVisible(appointment, filterValues)
      }));
    return data.filter((appointment) => this.isAppointmentVisible(appointment, filterValues));
  }

  private isAppointmentVisible(appointment: Appointment, filterValues: FilterValues) {
    const { startDate, endDate, query} = filterValues;
    const isExistByDate = new Date(appointment.startDate) >= startDate && new Date(appointment.endDate) <= endDate;
    const isExistByText = appointment.text.toLowerCase().includes(query);

    return isExistByDate && isExistByText;
  }
  ```